hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientReliableTopicProxy.java

```diff
@@ -118,6 +126,14 @@ public void publish(@Nonnull E payload) {
         }
     }
 
+    @Override
+    public CompletionStage<Void> publishAsync(@Nonnull E payload) {
+        checkNotNull(payload, NULL_MESSAGE_IS_NOT_ALLOWED);
+        Collection<E> messages = new ArrayList<>();
+        messages.add(payload);
+        return publishAllAsync(messages);
```

Comment:
Minor: you can use `Collections.singleton(payload)`.

Reply:
Fixed